zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -327,4 +319,55 @@ public static ResponseEntity<Object> mapDocumentHandlingExceptionToResponse(
       final DocumentException e) {
     return mapProblemToResponse(mapDocumentHandlingExceptionToProblem(e));
   }
+
+  public static ProblemDetail mapCamundaSearchExceptionToProblem(CamundaSearchException cse) {
+    final CamundaSearchException.Reason reason =
+        (cse.getReason() == null) ? CamundaSearchException.Reason.UNKNOWN : cse.getReason();
+    final Throwable cause = (cse.getCause() == null) ? cse : cse.getCause();
+    final String title =
+        (reason == CamundaSearchException.Reason.UNKNOWN)
+            ? cause.getClass().getName()
+            : reason.name();
+    final String errorMessage = cause.getMessage();
+    final String logPrefix = "Expected to handle REST request, but error occured: ";
+    switch (reason) {
+      case NOT_FOUND:
+        {
+          REST_GATEWAY_LOGGER.debug(logPrefix + "{}", errorMessage);
+          return createProblemDetail(
+              HttpStatus.NOT_FOUND, errorMessage, RejectionType.NOT_FOUND.name());
+        }
+      case NOT_UNIQUE:
+        {
+          REST_GATEWAY_LOGGER.debug(logPrefix + "{}", errorMessage);
+          return createProblemDetail(
+              HttpStatus.CONFLICT, errorMessage, RejectionType.INVALID_STATE.name());
+        }
+      case CONNECTION_FAILED:
+        {
+          final String detail = "The search client could to connect to the search server.";
+          REST_GATEWAY_LOGGER.debug(logPrefix + "{}", detail);
+          return createProblemDetail(HttpStatus.SERVICE_UNAVAILABLE, detail, title);
+        }
+      case SEARCH_SERVER_FAILED:
+        {
+          final String detail = "The search server was unable to process the request.";
+          REST_GATEWAY_LOGGER.debug(logPrefix + "{}", detail);
+          return createProblemDetail(HttpStatus.INTERNAL_SERVER_ERROR, detail, title);
+        }
+      case SEARCH_CLIENT_FAILED:
+        {
+          final String detail = "The search client was unable to process the request.";
```

Comment:
🔧 Let's keep the original message and add information.
```suggestion
          final String detail = errorMessage + ". The search client was unable to process the request.";
```